### PR TITLE
Updated "Increment Build Number" script

### PR DESCRIPTION
**Pull request by Sherlouk**
Relates to #356 

No more Info.plist merge conflicts 😱 

I've never tried pushing to Apple with a hash value as the build number, if they do cry about it we can always do the original script in the attached issue which uses the commit count for release builds.

### CODE REVIEW

**Comment by rnystrom** [comment]
Looks like it will probably get rejected 

https://developer.apple.com/library/content/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html#//apple_ref/doc/uid/20001431-102364

Another script:

https://gist.github.com/jpwatts/966838

From here 

https://stackoverflow.com/questions/8192921/set-xcode-bundle-version-to-git-hash

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Sherlouk** [comment]
The issue with `git rev-list HEAD --count` is that it's only at 892 😂 
Our Info.plist is over 1000 so we'd probably hit issues with conflicting/lower build numbers

**Comment by rizwankce** [comment]
Apple will reject if build number is fit hash and there won’t be any conflict if we increment a version. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by BasThomas** [comment]
I think you can reset the build number after every version though, so that shouldn’t be a problem @Sherlouk

**Comment by Sherlouk** [comment]
Pushed using commit count, if what @BasThomas has said is true then this should work for us going forward

**Comment by Sherlouk** [comment]
There's an irony to the conflicted file

**Comment by rnystrom** [comment]
@Sherlouk how about having the build # be this count, but then we include the hash as a custom key/value in the info plist? I really really like the idea of being able to tap into the release commit from settings, haha.

**Comment by Sherlouk** [comment]
@rnystrom Honestly I'd like to see us take more use of Releases on GitHub and correctly tag up when we push. Not only will we have a user-readable tag to link the user to, but we can attach release notes etc

This could be done automatically using fastlane

**Comment by rnystrom** [comment]
@Sherlouk agree, time to get more serious about these releases. That's why I setup the milestones too.

> This could be done automatically using fastlane

How so?

**Comment by Sherlouk** [comment]
> How so?

https://docs.fastlane.tools/actions/add_git_tag/
https://docs.fastlane.tools/actions/push_git_tags/

**Comment by BasThomas** [comment]
We should create an issue for that :)

File: Resources/Info.plist [hunk 1]
```diff
@@ -32,7 +32,7 @@
 		</dict>
 	</array>
 	<key>CFBundleVersion</key>
-	<string>1920</string>
+	<string>1</string>
```
Comment (rnystrom) [inline Resources/Info.plist]: @Sherlouk shouldn't this read 892 if the script works?

Comment (Sherlouk) [reply Resources/Info.plist]: Nope, the whole point of this script is that Resources/Info.plist **never changes**!

The actual compiled version of the app (found in your derived data) will have a different Info.plist with the correct value.

Again, the whole point of this change is to stop merge conflicts from every branch having a different build version.

**Comment by rnystrom** [comment]
Note: reverted for now, getting in the way too much from shipping TF builds atm. Will def try to revisit.

**Pull request merged**